File: japanesetest.rb
#!/usr/bin/ruby1.9
# encoding: utf-8

usedictionary = ARGV.include?("usedictionary")
noregen = ARGV.include?("noregen")
if !noregen
	if usedictionary
		system("./generatelexfile.rb usedictionary > japanese-dictionary.lex")
	else
		system("./generatelexfile.rb > japanese-dictionary.lex")
	end
end
system("cat japanese-rules.lex japanese-dictionary.lex > japanese.lex")
system("./generateyamlfile.rb > japanese.yaml")
system("./japanesetest.py")

File: japanesetest.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import sys
reload(sys)
sys.setdefaultencoding('utf-8')
import codecs
from kimmo import *

def listtostr(l):
	if type(l) == type([]):
		return "[" + (",".join([listtostr(x) for x in l])) + "]"
	elif type(l) == type((3,2)):
		return "(" + ",".join([listtostr(x) for x in l]) + ")"
	else:
		return l

k = KimmoRuleSet.load('japanese.yaml')
recfile = codecs.open('japanese.rec', encoding='utf-8')
for line in recfile:
    line = line.strip()
    if not line: continue
    if line.startswith(';'):
        print line
        continue
    featurelog = TextTrace(0)
    print listtostr(k.recognize(line, featurelog)), '<=', line
    print " ".join(featurelog.features)
    #print log

recfile.close()

